node/src/types/max_ttl.rs

```diff
@@ -35,6 +35,14 @@ impl MaxTtl {
         if highest_orphaned_block_header.is_genesis() {
             Ok(true)
         } else {
+            // It is possible that if a node just caught up, its highest orphaned block
+            // is within the current active era. We need to let the node sync back past the
+            // start of the active era and only after that we can determine if it is synced
+            // to TTL.
+            if latest_switch_block_timestamp < highest_orphaned_block_header.timestamp() {
+                return Ok(false);
+            }
+
```

Comment:
with era length of ~2hrs and TTL of greater than that I don't think this is necessary...so, am I misreading it or is this guard condition for a scenario where era length exceeds TTL or sth?

Reply:
Actually it was not the complete solve for the issue because we should determine an unbroken sequence of complete blocks from the start of the current era. The latest switch block header that was used may not have been from a complete block.
In the particular case of the sync test failure, the buddha block to which the node was first caught up was towards the middle of the last era. But since we did have the header of the last block of the previous era (got through sync leap or if the global state sync takes a long time to complete) we were comparing the old timestamp with the newer timestamp of the highest orphaned block which is the buddha block we just caught up to.

If we only check the TTL awareness using blocks from the complete range we don't have this issue anymore because if the last switch block is complete, it means that the highest orphaned block is either that switch block or a block that is older than it, so the previous check isn't needed anymore.

Fixed in this commit: https://github.com/casper-network/casper-node/pull/4190/commits/e0e41960028931f9d469c0bb142e4ca19239db68

Reply:
Lets discuss in standup tomorrow. I want to make sure I'm understanding the full picture before proceeding.